diff --git a/modules/aruco/samples/create_board.cpp b/modules/aruco/samples/create_board.cpp
index adb3a194..1ea4052e 100644
--- a/modules/aruco/samples/create_board.cpp
+++ b/modules/aruco/samples/create_board.cpp
@@ -59,6 +59,7 @@ const char* keys  =
         "{cd       |       | Input file with custom dictionary }"
         "{m        |       | Margins size (in pixels). Default is marker separation (-s) }"
         "{bb       | 1     | Number of bits in marker borders }"
+        "{id       | 0     | First id in the board }"
         "{si       | false | show generated image }";
 }
 
@@ -82,6 +83,7 @@ int main(int argc, char *argv[]) {
     }
 
     int borderBits = parser.get<int>("bb");
+    int id_0 = parser.get<int>("id");
     bool showImage = parser.get<bool>("si");
 
     String out = parser.get<String>(0);
@@ -115,7 +117,12 @@ int main(int argc, char *argv[]) {
         return 0;
     }
 
-    aruco::GridBoard board(Size(markersX, markersY), float(markerLength), float(markerSeparation), dictionary);
+    std::vector<int> ids; ids.reserve(markersX * markersY);
+    for (int id = 0; id < markersX * markersY; ++id) ids.push_back(id + id_0);
+    aruco::GridBoard board(Size(markersX, markersY),
+                           float(markerLength), float(markerSeparation),
+                           dictionary, ids);
+
 
     // show created board
     Mat boardImage;
